runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -120,7 +129,8 @@ public void write(IToken token, Object data) throws OverwriteException {
         });
 
         // Cache the successful write
-        if (!runtime.isCacheDisabled()) {
+        if ((opts == null || opts != null && opts.isDoCache() )
```

Comment:
simplify this conditional: ```!runtime.isCacheDisabled && (opts == null ||  opts.isCached())```
short-circuiting means second null check is not necessary